plugin/rename.py

```diff
@@ -172,39 +232,95 @@ def _get_relative_path(self, file_path: str) -> str:
         base_dir = wm.get_project_path(file_path)
         return os.path.relpath(file_path, base_dir) if base_dir else file_path
 
-    def _render_rename_panel(self, changes_per_uri: WorkspaceChanges, total_changes: int, file_count: int) -> None:
+    def _render_rename_panel(
+        self,
+        workspace_edit: WorkspaceEdit,
+        changes_per_uri: WorkspaceChanges,
+        total_changes: int,
+        file_count: int,
+        session_name: str
+    ) -> None:
         wm = windows.lookup(self.view.window())
         if not wm:
             return
-        panel = wm.panel_manager and wm.panel_manager.ensure_rename_panel()
+        pm = wm.panel_manager
+        if not pm:
+            return
+        panel = pm.ensure_rename_panel()
         if not panel:
             return
         to_render = []  # type: List[str]
+        reference_document = []  # type: List[str]
+        header_lines = "{} changes across {} files.\n\n".format(total_changes, file_count)
+        to_render.append(header_lines)
+        reference_document.append(header_lines)
+        ROWCOL_PREFIX = " {:>4}:{:<4} {}"
         for uri, (changes, _) in changes_per_uri.items():
             scheme, file = parse_uri(uri)
-            if scheme == "file":
-                to_render.append('{}:'.format(self._get_relative_path(file)))
-            else:
-                to_render.append('{}:'.format(uri))
+            filename_line = '{}:'.format(self._get_relative_path(file) if scheme == 'file' else uri)
+            to_render.append(filename_line)
+            reference_document.append(filename_line)
             for edit in changes:
-                start = parse_range(edit['range']['start'])
-                if scheme == "file":
-                    line_content = get_line(wm.window, file, start[0])
+                start_row, start_col_utf16 = parse_range(edit['range']['start'])
+                line_content = get_line(wm.window, file, start_row, strip=False) if scheme == 'file' else \
+                    '<no preview available>'
+                start_col = utf16_to_code_points(line_content, start_col_utf16)
```

Comment:
Seems like you've manage to discover some syntax highlighting bug:

<img width="693" alt="Screenshot 2024-03-10 at 22 12 30" src="https://github.com/sublimelsp/LSP/assets/153197/629fae71-e7dd-4020-8e35-6f10a3aaa227">

Reply:
It's already fixed upstream: https://github.com/sublimehq/Packages/commit/43268d1e726caf2ca78108de8e755741aa2a03e5